rholang/src/test/scala/coop/rchain/rholang/interpreter/PrettyPrinterTest.scala

```diff
@@ -582,7 +623,7 @@ class ProcPrinterSpec extends FlatSpec with Matchers {
       )
     result shouldBe
       """new x0, x1 in {
-        |  for( @{x2}, @{x3} <- @{x1} ; @{x4}, @{x5} <- @{x0} ) {
+        |  for( @{x2}, @{x3} <- @{*x1} ; @{x4}, @{x5} <- @{*x0} ) {
```

Comment:
It should say `*x6` down below. I spotted it by reading the comments above the AST creation code and seeing a `*d` there. Please make sure there are `*x` in assertions wherever there is `*x` in comments.

Reply:
Please add a separate minimal test for correct printing of `*x` in this scenario

Reply:
Guess you're right! We're always sending processes, and receiving names - so it is safe to match them against `@`-ed vars in the receive, so that the vars hold processes again.